[ETD-46] UART SEDS driver implementation

psurm opened the pull request.

psurm committed:
ETD-46] Initialize UART SEDS project with ETB components

Add hwas, assembler, piohwasdriver, uarhwasdriver functions
with connected interfaces between them. Use default implementation
from ETB without customization at this point.

Include project-range .gitignore which includes imported
functions.

psurm committed:
[ETD-46] Remove deploymentview

Deploymentview is non needed anymore. Values set before by DV are
available by default now. New Communication Device wizard does
not create DV anymore.

psurm committed:
[ETD-46] Copy piohwasdriver implementation from DemoSat2

psurm committed:
[ETD-46] Customize uarthwasdriver to meet UART SEDS criteria

Rename interfaces to match design criteria. Add needed XML variables.

psurm committed:
[ETD-46] Customize piohwasdriver to meet UART SEDS criteria

Rename interfaces to match design criteria. Add needed XML variables.

psurm committed:
[ETD-46] Update ocarina configuration with proper driver's types

Update AADL and Board1.xml snippets. Include proper driver's type name and install path.

psurm committed:
[ETD-46] Add install script

The script installs new driver in TASTE environment. It
merges system ocarina_components and Board1 with provided snippets.
It also copies generated driver files to tool-inst directory.
These steps make the driver available to use in target (standard)
TASTE project.

psurm committed:
[ETD-46] Add project name to interfaceview

The name is used in install script. New project wizard adds it by default.
Current project has been created with legacy wizard and we needed to add
the value manually.

psurm committed:
[ETD-46] Add UART configuration to ASN

psurm committed:
[ETD-46] Add dummy assembler implementation

Assembler translates data stream bytes between Broker and HWAS.
It is first dummy implementation, thad does not work yet, but
it is a first step in a research process.

psurm committed:
[ETD-46] Simplify driver name in install snippets

psurm committed:
[ETD-46] Update include path to Hal.h header

psurm committed:
[ETD-46] Remove function call from assert in HWAS

psurm committed:
[ETD-46] Fix IV settings (stack size and encoding)

Increase stack size for particular interfaces with the values from ETB
project. Some interfaces need more memory than other.

Change encoding to native, since at this point we do not support
ACN encoding.

psurm committed:
[ETD-46] Update install script

Recognize all needed ACNs and ASNs, generate needed files (ACN enc),
add asn1Scc prefix to all sources.

psurm committed:
[ETD-46] Implement one way translation between HWAS and Broker

psurm committed:
[ETD-46] Implement dummy provate data object

psurm committed:
[ETD-46] Implement translation between HWAS and Broker both ways

psurm committed:
[ETD-46] Add example usage of UART SEDS CD project

Lurkerpas commented on drivers/uart_seds/work/assembler/implem/default/C/src/assembler.c:
```diff
@@ -0,0 +1,107 @@
+                DECODED_PACKET_BUFFER_SIZE);
+
+   uart_seds_asn1SccUartHwasConfig config;
+   config.mInstance = UartHwas_Instance_uartHwas_Instance_3;
```
This should be read from the configuration.

psurm replied:
done

Lurkerpas commented on drivers/uart_seds/work/assembler/implem/default/C/src/assembler.c:
```diff
@@ -0,0 +1,107 @@
+   size_t index = 0;
+
+   Escaper_start_encoder(&escaper);
+   bytesToSend = Escaper_encode_packet(&escaper, data, length, &index);
```
It would be nice to explicitly set sentBytes  to 0 here

psurm replied:
done

Lurkerpas commented on drivers/uart_seds/uart_sedsconfiguration.asn:
```diff
@@ -0,0 +1,15 @@
+
+    UART-SEDS-Device-T    ::= ENUMERATED {uart0, uart1, uart2, uart3, uart4}
+
+    UART-SEDS-Baudrate-T  ::= ENUMERATED {b9600, b19200, b38400, b57600, b115200, b230400}
```
Some speeds are not supported by our device

psurm replied:
done

psurm committed:
[ETD-46] Update interface of driver model

Change name of 'data' field do distinguish it from private_data.
Add bus_id to Poll interface.

psurm committed:
[ETD-46] Use port instance and baudrate configured by user

psurm committed:
[ETD-161] Add support for multiple communication device

psurm committed:
[ETD-46] Initialize buffer iterator in Deliver function

psurm committed:
[ETD-46] Use actual bus_id in ReadByte UartHwas interface call

psurm committed:
[ETD-46] Remove explicit call to Broker from Assembler implemetation

Ficzer commented on drivers/uart_seds/interfaceview.ui.xml:
```diff
@@ -0,0 +1,134 @@
+<UI version="1.0" ><Entity id="{d8902a24-a304-4ddc-9e7a-cfd54f55b85d}"><Taste coordinates="9800 14200 81900 14600">
```
This file is not needed, add to .gitignore

psurm replied:
done

Ficzer commented on drivers/uart_seds/interfaceview.xml:
```diff
@@ -0,0 +1,425 @@
+</Provided_Interface>
+
+<Provided_Interface id="{14263882-5659-4760-8c52-cbf228553fa0}" name="InterruptManagement_DisableInterrupt_Pi" kind="Protected" enable_multicast="true" layer="default" required_system_element="NO" wcet="0" priority="1" is_simulink_interface="false" simulink_full_interface_ref="" dispatch_offset="0" stack_size="1024">
+    <Input_Parameter name="interrupt" type="InterruptNumber" encoding="NATIVE"/>
```
In this file, was encoding manually changed from ACN to NATIVE?

psurm replied:
Yes, added by the following patchset https://github.com/n7space/TASTE-CD-Demo/pull/1/commits/4e47ab1c47ff882edf46c4c446f90723fd8dff59

Ficzer commented on drivers/uart_seds/uart_seds.acn:
```diff
@@ -0,0 +1,11 @@
+UART-SEDS-DATAVIEW DEFINITIONS ::= BEGIN
+
+    --  In this ACN model, you can specify custom binary encoding rules for
```
Remove comments

psurm replied:
done

Ficzer commented on drivers/uart_seds/uart_sedsconfiguration.acn:
```diff
@@ -0,0 +1,10 @@
+UART-SEDS-DRIVER-CONFIGURATION DEFINITIONS ::= BEGIN
+    --  In this ACN model, you can specify custom binary encoding rules for
```
Remove comments

psurm replied:
done

Ficzer commented on drivers/uart_seds/uart_sedsprivatedata.acn:
```diff
@@ -0,0 +1,10 @@
+UART-SEDS-DRIVER-PRIVATE-DATAVIEW DEFINITIONS ::= BEGIN
+    --  In this ACN model, you can specify custom binary encoding rules for
```
Remove comments

psurm replied:
done

Ficzer commented on drivers/uart_seds/work/assembler/implem/default/C/src/assembler.c:
```diff
@@ -0,0 +1,156 @@
+static uint64_t bytesToSend;
+
+__attribute__((section(".sdramMemorySection")))
+static Escaper escaper;
```
Wasn't this part a short term sollution? Shouldn't it be placed in private data?

psurm replied:
It should be, however at this point we decided to store all the above vars as globals inside assembler. It will be polished when we implement RS785

Ficzer commented on drivers/uart_seds/Makefile:
```diff
@@ -0,0 +1,97 @@
+install:
+	chmod 755 install.sh
+	./install.sh
+	$(COMMUNICATION_DEVICE_INSTALLATOR) -c ${COMMUNICATION_DEVICE_PATH}
```
As a side note, this line will need to be updated after my MRs merges

Ficzer commented on targets/uart_usage_test/deploymentviewstandard.ui.xml:
```diff
@@ -0,0 +1,17 @@
+<UI version="1.0" ><Entity id="{7520c1db-3052-4b3f-b26b-be22801b9814}"><Taste coordinates="32200 30000 48200 46000">
```
This file can be removed

psurm replied:
done

Ficzer commented on targets/uart_usage_test/interfaceview.ui.xml:
```diff
@@ -0,0 +1,20 @@
+<UI version="1.0" ><Entity id="{697d9395-eeb7-4344-9878-86b87c0aee57}"><Taste coordinates="31200 33000 51200 41000">
```
This file can be remove

psurm replied:
done

Ficzer commented on targets/uart_usage_test/uart_usage_test.acn:
```diff
@@ -0,0 +1,12 @@
+UART-USAGE-TEST-DATAVIEW DEFINITIONS ::= BEGIN
+
+    --  In this ACN model, you can specify custom binary encoding rules for
```
Remove comment

psurm replied:
done

Ficzer commented on targets/uart_usage_test/uart_usage_test.asn:
```diff
@@ -0,0 +1,42 @@
+UART-USAGE-TEST-DATAVIEW DEFINITIONS ::=
+BEGIN
+
+   MyInteger ::= INTEGER (0 .. 10000)
```
Remove comment and unnecessary asn definition

psurm replied:
comments removed, but that definition is used in implemetantion

psurm committed:
[ETD-46] Clean projects, remove unused commented out code and IDE files
